layer/contracts/definitions.py

```diff
@@ -147,15 +124,58 @@ def package(self) -> Path:
     def _package_executable(self) -> Path:
         resource_paths = [Path(resource.path) for resource in self.resource_paths]
         return package_function(
-            self.func,
+            self.runner_function(),
             resources=resource_paths,
-            assertions=self.assertions,
             pip_dependencies=self.pip_dependencies,
             output_dir=self.function_home_dir,
         )
 
+    def runner_function(self) -> Any:
+        if self.asset_type == AssetType.DATASET:
+            from layer.executables.dataset.entrypoint import runner as dataset_runner
+
+            return dataset_runner(self)
+        elif self.asset_type == AssetType.MODEL:
+            from layer.executables.model.entrypoint import runner as model_runner
+
+            return model_runner(self)
+        raise Exception(f"Invalid asset type {self.asset_type}")
+
+    def _clean_function_home_dir(self) -> None:
+        # Remove directory to clean leftovers from previous runs
+        function_home_dir = self.function_home_dir
+        if function_home_dir.exists():
+            shutil.rmtree(function_home_dir)
+        os.makedirs(function_home_dir)
```

Comment:
is this acting in a tmp location? or can it wipe the user files if they use it wrongly?

Reply:
this is not new code, just organizing existing methods into deprecated and regular methods